pkg/apis/core/validation/secretbinding.go

```diff
@@ -69,6 +70,16 @@ func validateSecretReferenceOptionalNamespace(ref corev1.SecretReference, fldPat
 
 	if len(ref.Name) == 0 {
 		allErrs = append(allErrs, field.Required(fldPath.Child("name"), "must provide a name"))
+	} else {
+		for _, err := range validation.IsDNS1123Subdomain(ref.Name) {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("name"), ref.Name, err))
+		}
+	}
+
+	if len(ref.Namespace) > 0 {
+		for _, err := range validation.IsDNS1123Subdomain(ref.Namespace) {
```

Comment:
We could directly use the upstream function that validates Namespace name: [`apivalidation.ValidateNamespaceName`](https://github.com/kubernetes/kubernetes/blob/v1.34.0/staging/src/k8s.io/apimachinery/pkg/api/validation/generic.go#L60-L63):

```suggestion
		for _, err := range apivalidation.ValidateNamespaceName(ref.Namespace, false) {
```